applications/vanilla/controllers/class.discussioncontroller.php

```diff
@@ -82,7 +82,9 @@ public function index($DiscussionID = '', $DiscussionStub = '', $Page = '') {
         // Check permissions.
         $Category = CategoryModel::categories($this->Discussion->CategoryID);
         $this->permission('Vanilla.Discussions.View', true, 'Category', val('PermissionCategoryID', $Category, -1));
-        $this->setData('CategoryID', $this->CategoryID = $this->Discussion->CategoryID, true);
+        if (c('Vanilla.Categories.Use', true)) {
+            $this->setData('CategoryID', $this->CategoryID = $this->Discussion->CategoryID, true);
```

Comment:
`$this->setData('CategoryID', $this->CategoryID);`
since `$this->CategoryID` is already defined